common/multistep_runner.go

```diff
@@ -0,0 +1,156 @@
+package common
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/mitchellh/multistep"
+	"github.com/mitchellh/packer/packer"
+)
+
+func newRunner(steps []multistep.Step, config PackerConfig, ui packer.Ui) (multistep.Runner, multistep.DebugPauseFn) {
+	switch config.PackerOnError {
+	case "cleanup":
+	case "abort":
+		for i, step := range steps {
+			steps[i] = abortStep{step, ui}
+		}
+	case "ask":
+		for i, step := range steps {
+			steps[i] = askStep{step, ui}
+		}
+	default:
+		ui.Error(fmt.Sprintf("Ignoring -on-error=%q argument: unknown on-error value", config.PackerOnError))
```

Comment:
I have already deleted the default clause since it can no longer happen. Now `enumflag` makes packer exit with an error like `invalid value "ASK" for flag -on-error: expected one of ["cleanup" "abort" "ask"]`